src/server/modules/threat/services/SafeBrowsingService.ts

```diff
@@ -89,6 +90,76 @@ export class SafeBrowsingService implements UrlThreatScanService {
     }
     return matches
   }
+
+  // TODO: deduplicate this function with the lookup function above
+  // ideally, we should do a non-block write for any matched viruses
+  // into the redis cache. the write above is blocking so we chose to
+  // bypass the write for now
```

Comment:
Discussion: can we just call the `set` function without `await`, technically, will turn it into a non-block call. I did this all the time in other languages like: call a logging function or write/invalidate some cache that I don't need to wait till it completes.
The only bad thing that can happen is the function might return an error that isn't caught properly 🤔 
Ref: https://stackoverflow.com/questions/61834275/what-happens-to-the-promise-if-we-dont-await-for-it

Reply:
Good idea, I opened as separate issue for refactoring because I think the error handling will need to be tested more rigorously to prevent any errors from bubbling up and affecting the application

Reply:
#1996